3 - Assignment/240521 - Rust Basic + Roguelike Game, Week 4/binary-tree/src/lib.rs

```diff
@@ -20,6 +20,65 @@ pub struct BinaryTree<T: Ord> {
 
 // Implement `new`, `insert`, `len`, and `has`.
 
+impl<T: Ord> Subtree<T> {
+    pub fn len(&self) -> i32 {
+        if self.0.is_none() {
+            return 0;
+        }
+        let node = self.0.as_ref().unwrap();
+        node.left.len() + node.right.len() + 1
```

Comment:
`match` 문을 활용한 패턴 매칭으로 리팩토링하면 좀 더 깔끔해집니다.
```suggestion
        match &self.0 {
            Some(n) => 1 + n.left.len() + n.right.len(),
            None => 0,
        }
```